File: gridaj.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

int main() 
{
FILE *output;
double *xi;
double *gridx,*gridy;
int i,j;
int n = 4;
double a = -1;
double b = 1;
double h = (b - a) /(n-1);

xi = (double *) malloc(n*sizeof(double));
gridx = (double *) malloc(n*n*sizeof(double));
gridy = (double *) malloc(n*n*sizeof(double));

for (i = 0; i < n; i++) {
    xi[i] = a + i*h;
    printf("xi[%d] = %f \n",i,xi[i]);
    }

for (i = 0; i < n*n; i++){
    for (j = 0; j < n; j++){
        gridx[i*n+j] = xi[j];
        gridy[j*n+i] = xi[j];  
  }
}
  
/*for (i = 0; i < n*n; i++){*/
/*    for (j = 0; j < n; j++){*/
/*        printf("xij[%d] = %f \n",i*n+j,gridx[i*n+j]);*/
/*        printf("xji[%d] = %f \n",j*n+i,gridy[i*n+j]);*/
/*    }*/
/*}*/

output = fopen ( "file.dat", "wt" );
        
if ( !output ){
    fprintf ( stderr, "\n" );
    fprintf ( stderr, "WRITE_RESULTS - Fatal error!\n" );
    fprintf ( stderr, "  Could not open the output file.\n" );
    exit ( 1 );
}

//Write the data.
for ( i = 0; i < n*n; i++ )
for ( j = 0; j < n; j++ ){
    fprintf ( output, "  %24.16g\t%24.16g\t", gridx[i*n+j], gridy[i*n+j]);
}
    return 0;
}

File: gridaj.py
def gridaj(a,b,nx,ny):
    hx=(b-a)/(nx-1);hy=(b-a)/(ny-1)
    x=np.zeros(nx)
    y=np.zeros(ny)
    gridx=np.zeros((nx)**2)
    gridy=np.zeros((ny)**2)
    for i in range(0,nx):
        x[i] = a + i*hx
        y[i] = a + i*hy
    for i in range(0,nx):  
        for j in range(0,nx):
            gridx[i*(nx)+j] = x[j]
    
    for i in range(0,ny):  
        for j in range(0,ny):
            gridy[j*(ny)+i] = y[j]

    return gridx,gridy
